packages/x-date-pickers/src/DateTimePicker/DateTimePickerToolbar.tsx

```diff
@@ -235,10 +234,15 @@ const DateTimePickerToolbarAmPmSelection = styled('div', {
 });
 
 /**
- * If this context value is set to true, the toolbar will always be rendered in the desktop mode.
+ * If `forceDesktopVariant` is set to true, the toolbar will always be rendered in the desktop mode.
+ * If `onViewChange` is defined, the toolbar will call it instead of calling the one from `usePickerContext`.
  * This is used by the Date Time Range Picker Toolbar.
  */
-export const DateTimePickerToolbarForceDesktopVariant = React.createContext(false);
+export const DateTimePickerToolbarOverrideContext = React.createContext<{
+  forceDesktopVariant: boolean;
+  onViewChange: (view: DateOrTimeViewWithMeridiem) => void;
+  view: DateOrTimeViewWithMeridiem | null;
```

Comment:
is there a way we can remove the ` | null` here?

Reply:
Nop, the `DateTimeRangePickerToolbar` passes `null` when you are editing the other date